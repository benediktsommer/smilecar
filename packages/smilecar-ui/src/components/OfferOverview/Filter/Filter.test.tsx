import { renderHook, act } from '@testing-library/react-hooks';
import { render } from '@testing-library/react';

import { Filter } from './';

const mockedFilterValue = jest.fn();
jest.mock('./context', () => ({
  useFilterState: () => ({
    addFilter: () => {
      /* do nothing */
    },
    deleteFilter: () => {
      /* do nothing */
    },
    filter: mockedFilterValue(),
  }),
}));

const mockedOriginalFilter = jest.fn();
jest.mock('../../../hooks/useGetFilter', () => ({
  useGetFilter: () => ({
    filter: mockedOriginalFilter(),
  }),
}));

const mockedAvailableFilter = jest.fn();
jest.mock('../../../hooks/useGetAvailableFilter', () => ({
  useGetAvailableFilter: () => ({
    availableFilter: mockedAvailableFilter(),
  }),
}));

describe('test <Filter />', () => {
  const renderComponent = () => render(<Filter />);

  beforeEach(() => {
    mockedFilterValue.mockReset();
    mockedOriginalFilter.mockReset();
    mockedAvailableFilter.mockReset();
  });

  describe('if the filter options are loaded', () => {
    beforeEach(() => {
      mockedAvailableFilter.mockReturnValue({
        seats: [2, 4, 6],
      });

      mockedOriginalFilter.mockReturnValue({
        seats: [4, 5],
      });

      mockedFilterValue.mockReturnValue({
        selected: {
          seats: [4],
        },
      });
    });

    it('should display the filter option', () => {
      const { asFragment } = renderComponent();
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('if the loaded filter options are empty', () => {
    beforeEach(() => {
      mockedAvailableFilter.mockReturnValue(null);

      mockedOriginalFilter.mockReturnValue(null);

      mockedFilterValue.mockReturnValue({});
    });

    it('should display nothing', () => {
      const { asFragment } = renderComponent();
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
