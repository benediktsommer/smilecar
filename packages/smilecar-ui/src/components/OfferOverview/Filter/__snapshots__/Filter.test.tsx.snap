// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <Filter /> if the filter options are loaded should display the filter option 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-2 Styled(MuiBox)-root-1"
    classes="[object Object]"
  >
    <div
      class="MuiPaper-root MuiAccordion-root WithStyles(ForwardRef(Accordion))-root-3 MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        aria-controls="panel1a-content"
        aria-disabled="false"
        aria-expanded="false"
        class="MuiButtonBase-root MuiAccordionSummary-root"
        id="panel1a-header"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            SEATS
          </p>
        </div>
        <div
          aria-disabled="false"
          aria-hidden="true"
          class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        class="MuiCollapse-container MuiCollapse-hidden"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              aria-labelledby="panel1a-header"
              id="panel1a-content"
              role="region"
            >
              <div
                class="MuiAccordionDetails-root WithStyles(ForwardRef(AccordionDetails))-root-4"
              >
                <div
                  class="MuiBox-root MuiBox-root-7 makeStyles-filterOption-5 makeStyles-filterOption-6"
                >
                  <p>
                    4
                  </p>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-9 makeStyles-filterOption-5 makeStyles-filterOption-8"
                >
                  <p>
                    5
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-10 MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Filter zur√ºcksetzen
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`test <Filter /> if the loaded filter options are empty should display nothing 1`] = `<DocumentFragment />`;
